test/gtest/multilabelsoftmarginloss.hpp

```diff
@@ -0,0 +1,275 @@
+                                                                  target.desc,
+                                                                  target_dev.get(),
+                                                                  weight.desc,
+                                                                  weight_dev.get(),
+                                                                  output.desc,
+                                                                  output_dev.get());
+
+        EXPECT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+        auto error       = miopen::rms_range(ref_output, output);
+        EXPECT_TRUE(miopen::range_distance(ref_output) == miopen::range_distance(output));
+        EXPECT_TRUE(error < threshold) << "Error output beyond tolerance "
+                                          "Error:"
+                                       << error << ",  Threshold: " << threshold;
+    }
+    MultilabelSoftMarginLossTestCase config;
+
+    tensor<T> input;
+    tensor<T> target;
+    tensor<T> weight;
+    tensor<T> output;
+
+    tensor<T> ref_output;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr target_dev;
+    miopen::Allocator::ManageDataPtr weight_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+};
+
+template <typename T = float>
+struct MultilabelSoftMarginLossReducedForwardTest
+    : public ::testing::TestWithParam<MultilabelSoftMarginLossTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        config        = GetParam();
+
+        auto in_dims = config.dims;
+
+        auto gen_in_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(0.1, 50); };
+        input             = tensor<T>{in_dims};
+        std::generate(input.begin(), input.end(), gen_in_value);
+
+        auto gen_target_value = [](auto...) { return prng::gen_A_to_B<int32_t>(0, 2); };
+        target                = tensor<T>{in_dims};
+        std::generate(target.begin(), target.end(), gen_target_value);
+
+        auto gen_weight_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(0.1, 50); };
+        // input is (N, C) -> weight is (C)
+        weight = tensor<T>{std::vector<size_t>{in_dims[1]}};
+        std::generate(weight.begin(), weight.end(), gen_weight_value);
+
+        // Tensor with 1 element to store result after reduce
+        output = tensor<T>{std::vector<size_t>{1}};
+        std::fill(output.begin(), output.end(), 0);
+
+        ref_output = tensor<T>{std::vector<size_t>{1}};
+        std::fill(ref_output.begin(), ref_output.end(), 0);
+
+        // Change this to MIOPEN_LOSS_REDUCTION_SUM to test sum reduction
+        reduction_mode = MIOPEN_LOSS_REDUCTION_MEAN;
```

Comment:
It seems better to work on this part by receiving it from the MultilabelSoftMarginLossTestConfigs function.

Reply:
Fixed in [this commit](https://github.com/moreh-dev/MIOpen/pull/33/commits/f4ae3458f5a32f213d8f94c64d822ea2696f6f8a)